app/src/main/java/com/kylecorry/trail_sense/navigation/ui/NavigatorFragment.kt

```diff
@@ -197,25 +195,50 @@ class NavigatorFragment : Fragment() {
             true
         }
 
-        binding.navigationOpenArCamera.setOnClickListener { view ->
+        binding.navigationOpenArCamera.setOnClickListener {
             if (!viewCameraBindToLifecycle) {
                 binding.viewCamera.bindToLifecycle(viewLifecycleOwner)
                 viewCameraBindToLifecycle = true
             }
             binding.viewCameraLine.visibility = if (binding.viewCamera.visibility == View.VISIBLE) View.INVISIBLE else View.VISIBLE
             binding.viewCamera.visibility = if (binding.viewCamera.visibility == View.VISIBLE) View.INVISIBLE else View.VISIBLE
+            if (userPrefs.navigation.rightQuickAction == QuickActionType.Flashlight) {
+                    binding.navigationRightQuickAction.isClickable = if (binding.viewCamera.visibility == View.VISIBLE) false else true
+                }
+            if (userPrefs.navigation.leftQuickAction == QuickActionType.Flashlight) {
+                binding.navigationLeftQuickAction.isClickable = if (binding.viewCamera.visibility == View.VISIBLE) false else true
+            }
+            sightingCompassOpen = if (binding.viewCamera.visibility == View.VISIBLE) true else false
+            sightingCompassOpen = if (binding.viewCamera.visibility == View.VISIBLE) true else false
+            sightingCompassLineEnabled = if (binding.viewCameraLine.visibility == View.VISIBLE) true else false
         }
 
         binding.navigationOpenArCamera.setOnLongClickListener {
```

Comment:
In my testing, I didn't really find that the long click functionality provided much use - is there a reason you can think of to hide the sighting line?

Reply:
No, not really. I can remove it, so the code is cleaner.